app/uk/gov/hmrc/cardpaymentfrontend/views/PaymentCompletePage.scala.html

```diff
@@ -0,0 +1,94 @@
+@*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *@
+
+@import uk.gov.hmrc.cardpaymentfrontend.views.html.Layout
+@import uk.gov.hmrc.govukfrontend.views.Aliases.{Key, SummaryList, Text, Value}
+@import uk.gov.hmrc.govukfrontend.views.html.components.{GovukPanel, GovukSummaryList, Panel}
+@import uk.gov.hmrc.govukfrontend.views.viewmodels.content.HtmlContent
+@import uk.gov.hmrc.govukfrontend.views.viewmodels.summarylist.SummaryListRow
+@import uk.gov.hmrc.cardpaymentfrontend.models.EmailAddress
+
+@this(
+        govukPanel: GovukPanel,
+        govukSummaryList: GovukSummaryList,
+        layout: Layout
+)
+
+@(
+        taxReference: String,
+        taxType: String,
+        amountInPounds: String,
+        paymentDate: String,
+        maybeEmailAddress: Option[EmailAddress]
+)(implicit request: RequestHeader, messages: Messages)
+
+@heading = @{messages("payment-complete.panel.title")}
+
+@panelContent = @{Html(messages("payment-complete.panel.content", taxReference))}
+
+@maybeEmailContent = {
+    @maybeEmailAddress.fold(Html("")) { emailAddress =>
+        <p id="email-paragraph" class="govuk-body">
+            @Html(Messages("payment-complete.email.paragraph", emailAddress.value))
+        </p>
+    }
+}
+
+@layout(pageTitle = Some(heading), showBackLink = false) {
+
+    @govukPanel(Panel(
+        title = Text(heading),
+        content = HtmlContent(panelContent)
+    ))
+
+    @maybeEmailContent
+
+    @govukSummaryList(SummaryList(
```

Comment:
Could this be injected?

Reply:
Do you mean built as a partial in another file? Or passed in as an arg to the contractor?

It opens an interesting conversation as to whether we build components for the twirl templates in scala files (like in some places in pay-frontend) or it's built within the twirl template (also, done in some places in pay-frontend).